sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/ResourceBase.java

```diff
@@ -34,6 +37,46 @@
   protected static final BridgeProtoValueConverters PROTO_CONVERTERS =
       BridgeProtoValueConverters.instance();
 
+  // // // Helper methods for Schema access
+
+  /**
+   * Method to call to try to access Table metadata and call given Function with it (if successful),
+   * or, create and return an appropriate error Response if access fails.
+   */
+  protected Response callWithTable(
+      StargateGrpc.StargateBlockingStub blockingStub,
+      String keyspaceName,
+      String tableName,
+      Function<Schema.CqlTable, Response> function) {
+    final Schema.CqlTable tableDef;
+    try {
+      tableDef = BridgeSchemaClient.create(blockingStub).findTable(keyspaceName, tableName);
+    } catch (Exception e) {
+      if (e instanceof StatusRuntimeException) {
+        StatusRuntimeException grpcE = (StatusRuntimeException) e;
+        switch (grpcE.getStatus().getCode()) {
+          case NOT_FOUND:
+            final String msg = e.getMessage();
+            if (msg.contains("Keyspace not found")) {
+              return jaxrsBadRequestError(String.format("Keyspace '%s' not found", keyspaceName))
+                  .build();
+            }
+            return jaxrsBadRequestError(
+                    String.format(
+                        "Table: '%s' not found (in keyspace %s)", tableName, keyspaceName))
+                .build();
+        }
```

Comment:
would this expose a possibility to actually get the understanding if there is an existing keyspace/table information without actually being authorized to do so?

Reply:
There is a bigger discussion to be had about access, yes. But on short term gRPC API does check access rights AFAIK. There will be challenges in caching wrt tenancy.

Handling here is meant to replace SGv1 checks and produce same HTTP response codes, FWTW.
And try to extract it into shared handling, away from endpoint implementation.